provisioning/baremetal_config.go

```diff
@@ -0,0 +1,84 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/golang/glog"
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+const (
+	// BaremetalProvisioningCR is the name of the provisioning config resource.
+	BaremetalProvisioningCR = "provisioning-configuration"
+)
+
+// ValidateBaremetalProvisioningConfig validates the contents of the baremetal Provisioning CR.
+func ValidateBaremetalProvisioningConfig(prov *metal3iov1alpha1.Provisioning) error {
+	provisioningNetworkMode := getProvisioningNetworkMode(prov)
+	switch provisioningNetworkMode {
+	case metal3iov1alpha1.ProvisioningNetwork("Managed"):
```

Comment:
You can refer to the const values directly.

```suggestion
	case metal3iov1alpha1.Managed:
```

That may mean that we want to make those names more verbose. `ProvisioningNetworkManaged`, for example.

Reply:
Done.